src/hooks/UrqlUseSubscription.re

```diff
@@ -0,0 +1,59 @@
+[@bs.deriving abstract]
+type useSubscriptionArgs = {
+  query: string,
+  [@bs.optional]
+  variables: Js.Json.t,
+};
+
+[@bs.deriving abstract]
+type useSubscriptionResponseJs = {
+  fetching: bool,
+  [@bs.optional]
+  data: Js.Json.t,
+  [@bs.optional]
+  error: UrqlCombinedError.t,
+};
+
+type useSubscriptionResponse('response) = {
+  fetching: bool,
+  data: option('response),
+  error: option(UrqlCombinedError.t),
+  response: UrqlTypes.response('response),
+};
+
+[@bs.module "urql"]
+external useSubscriptionJs:
+  useSubscriptionArgs => array(useSubscriptionResponseJs) =
```

Comment:
I believe it's possible do to away with `useSubscriptionWithHandlerJs` if we change the signature here and below with something like: `(~handler: UrqlTypes.handler('acc)=?, useSubscriptionArgs) => array(useSubscriptionResponseJs)`

I'm working on an example of how it could work

Reply:
Ah gotcha, so we'd have one `useSubscriptionJs` that takes an optional argument for `~handler` for when `useSubscriptionWithHandler` binds to it. I could see that working!